### ISSUE TITLE
MenuItem registerAccelerator does not work

### ISSUE THREAD

**Issue by Maxime117** [electron/electron#18295]
### Preflight Checklist

* [x] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/master/CONTRIBUTING.md) for this project.
* [x] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/master/CODE_OF_CONDUCT.md) that this project adheres to.
* [x] I have searched the issue tracker for an issue that matches the one I want to file, without success.

### Issue Details

* **Electron Version:**
  * 5.0.1
* **Operating System:**
  * macOS 10.14.4

### Expected Behavior
Hello,

For some reasons, I have to use the window.keydown method instead of the MenuItem accelerator, but I still want to display the shortcut to the right of the MenuItem. So I used 'registerAccelerator: false'. But the click() function is always called when I hit "B".

### To Reproduce
```
{
  label: "Blade",
  id: "blade",
  accelerator: "B",
  registerAccelerator: false,
  click(a, b, e) {
    console.log(e);
  }
}
```

**Comment by codebytere** [comment]
@Maxime117 did this work in any previous versions?

**Comment by Maxime117** [comment]
@codebytere It didn't work in v4.x.x

**Comment by DeerMichel** [comment]
So me and @erickzhao played around with this issue and at least on Mac we're kinda stuck because of the limited `NSMenuItem` API. Setting the accelerator also causes the shortcut label to be shown and not setting the accelerator (to prevent it from firing the click handler) does result in not showing the label. There is no "Apple" way to separate those two things.

I fiddled around with Xcode and came to a possible solution for displaying a shortcut label without setting the accelerator using a custom view for `NSMenuItem` (containing 2 labels: menu label aligned left, shortcut label aligned right):
![image](https://user-images.githubusercontent.com/8472721/59538877-11785d80-8eb0-11e9-9d7c-6c34890a46f7.png)
Problem is, this involves reengineering the whole `NSMenuItem` to match style and behavior of the native view. Probably too much effort for an edge case like this.

_Side note_: we didn't investigate how the property behaves on Linux or Windows.

**TL;DR** two options:
- remove the `registerAccelerator` property; though someone needs to check how this affects #15719 because it reverts the fix from #15723
- implement custom view on Mac and adapt other platforms as needed

cc @codebytere

**Comment by codebytere** [comment]
> Problem is, this involves reengineering the whole NSMenuItem to match style and behavior of the native view. Probably too much effort for an edge case like this.

In an ideal world we don't want to introduce cross-platform incompatibilities but there's also a clear question of maintainer bandwidth and in my opinion the maintenance burden of introducing a custom view outweighs the cost to consumers of not being able to see a hotkey but not have it work as a traditional accelerator. So i'll second your estimation that that option isn't worthwhile, but i don't necessarily think that, conversely, the asymmetry is worth taking functionality _away_ from users on other platforms, so I think the right path forward here might be to indicate more clearly to users that `registerAccelerator` is a Linux and Mac-only property until either we can find a better solution or until Apple allows us greater NSMenuItem granularity.

**Comment by ronen** [comment]
FYI, in addition to `registerAccelerator: false` not working it seems that `event.triggeredByAccelerator` isn't being set.  I don't know if that's due to this same issue or something different.

**Comment by codebytere** [comment]
@ronen that'll be resolved by https://github.com/electron/electron/pull/18865 :)

**Comment by DeerMichel** [comment]
It's also worth mentioning that the `event.triggeredByAccelerator` flag was only introduced in version 6 (currently in beta).

**Comment by DeerMichel** [comment]
`registerAccelerator: false` confirmed to work under Windows & Linux (Electron v6)

**Comment by inDigiNeous** [comment]
Can you please clarify this in the documentation that this does not work on Apple currently. 
https://electronjs.org/docs/api/menu-item does not say anything about not working on macOS. Thanks.
